### ISSUE TITLE
[Bug] CSRF token invalid with 0.15.1

### ISSUE THREAD

**Issue by badstorm** [gobuffalo/buffalo#1843]
# Description
If you start a new project with an empty go home directory you get `CSRF token invalid` in any form implementation.

On the Buffalo error page in Context i get

`authenticity_token: mDgs2wusjsui98vp/1w7p48RW5tds2+wFUJUQXcIjnMzHXffTweESj7dUNixg7CM7UlFojYMkFPWJ3Pr6BJH7g==`

but the Form section i get

`map[Title:[adsfg] authenticity_token:[3eKcz3ub9ATh7zQMbhJnsyoUmfdql9DjR5qvZI7oMcirRVxjQqXrs61zbzOReUWdGyPPA1+oW3NyAnOPF/MXSg==]]`

I tried this on differnt computers with the same result.

### Actual Behavior

- install buffalo cli 0.15.1
- Init and empty go home
- in src create a new buffalo project with `buffalo new`
- add a resource with `buffalo g r `
- test the form

### Info

Please run `buffalo info` and paste the information below where it says "PASTE_HERE".

<details>

```
-> Go: Checking installation
✓ The `go` executable was found on your system at: /usr/local/go/bin/go

-> Go: Checking minimum version requirements
✓ Your version of Go, 1.13, meets the minimum requirements.

-> Go: Checking GOPATH
✓ You appear to be operating inside of your GOPATH.

-> Go: Checking Package Management
⚠ You do not appear to be using a package management system.

It is strongly suggested that you use one of the following package management systems:

* Go Modules (Recommended) - https://gobuffalo.io/en/docs/gomods
* Dep - https://github.com/golang/dep

For help setting up your Go environment please follow the instructions for you platform at:

https://www.gopherguides.com/courses/preparing-your-environment-for-go-development

-> Go: Checking PATH
✓ Your PATH contains /Users/badstorm/Source/GO/bin.

-> Node: Checking installation
✓ The `node` executable was found on your system at: /usr/local/bin/node

-> Node: Checking minimum version requirements
✓ Your version of Node, v12.12.0, meets the minimum requirements.

-> NPM: Checking installation
✓ The `npm` executable was found on your system at: /usr/local/bin/npm

-> NPM: Checking minimum version requirements
✓ Your version of NPM, 6.11.3, meets the minimum requirements.

-> Yarn: Checking installation
✓ The `yarnpkg` executable was found on your system at: /usr/local/bin/yarnpkg

-> Yarn: Checking minimum version requirements
✓ Your version of Yarn, 1.19.1, meets the minimum requirements.

-> PostgreSQL: Checking installation
✘ The `postgres` executable could not be found on your system.
For help setting up your Postgres environment please follow the instructions for you platform at:

https://www.postgresql.org/download/

-> MySQL: Checking installation
✘ The `mysql` executable could not be found on your system.
For help setting up your MySQL environment please follow the instructions for you platform at:

https://www.mysql.com/downloads/

-> SQLite3: Checking installation
✓ The `sqlite3` executable was found on your system at: /usr/bin/sqlite3

-> SQLite3: Checking minimum version requirements
✓ Your version of SQLite3, 3.28.0, meets the minimum requirements.

-> Cockroach: Checking installation
✘ The `cockroach` executable could not be found on your system.
For help setting up your Cockroach environment please follow the instructions for you platform at:

https://www.cockroachlabs.com/docs/stable/

-> Buffalo (CLI): Checking installation
✓ The `buffalo` executable was found on your system at: /usr/local/bin/buffalo

-> Buffalo (CLI): Checking minimum version requirements
✓ Your version of Buffalo (CLI), v0.15.1, meets the minimum requirements.

-> Buffalo: Application Details
Pwd         /Users/badstorm/Source/GO/src/duckpage.net/std01
Root        /Users/badstorm/Source/GO/src/duckpage.net/std01
GoPath      /Users/badstorm/Source/GO
PackagePkg  duckpage.net/std01
ActionsPkg  duckpage.net/std01/actions
ModelsPkg   duckpage.net/std01/models
GriftsPkg   duckpage.net/std01/grifts
WithModules false
Name        std01
Bin         bin/std01
VCS         git
WithPop     true
WithSQLite  false
WithDep     false
WithWebpack true
WithNodeJs  true
WithYarn    true
WithDocker  true
WithGrifts  true
AsWeb       true
AsAPI       false
InApp       true
PackageJSON {map[build:webpack -p --progress dev:webpack --watch]}

-> Buffalo: config/buffalo-app.toml
name = "std01"
bin = "bin/std01"
vcs = "git"
with_pop = true
with_sqlite = false
with_dep = false
with_webpack = true
with_nodejs = true
with_yarn = true
with_docker = true
with_grifts = true
as_web = true
as_api = false

-> Buffalo: config/buffalo-plugins.toml
[[plugin]]
  binary = "buffalo-pop"
  go_get = "github.com/gobuffalo/buffalo-pop"

```

</details>

**Comment by rpwatkins** [comment]
I can confirm the same happens with an existing project. Upgrading to 0.15.1 caused the same CSRF toekn invalid error.

**Comment by cmorbitzer** [comment]
This is caused by #1832, which enables the HttpOnly and Secure cookie attributes. This is a problem if your app runs over HTTP, e.g. in development.

**Comment by paganotoni** [comment]
@cmorbitzer I've tried to reproduce it but have not been able to. I've tested the app in development but so far POST requests work. I'll keep working on it and keep you posted.

**Comment by cmorbitzer** [comment]
@paganotoni It happens for me if I clear my cookies.

It doesn't happen if the `_roleboard_session` cookie is already set on the browser. However, if you clear that cookie it will not be written again over HTTP unless the line `cookieStore.Options.Secure = true` is removed from `options.go`. So the CSRF token will never be valid because the session cookie cannot be set on the browser.

**Comment by paganotoni** [comment]
Thanks @cmorbitzer !

**Comment by paganotoni** [comment]
Solved in #1851 and release in v0.15.2

**Comment by boonsanti** [comment]
Got "CSRF token invalid" after upgrade to version 0.15.5

buffalo info >
`
-> Go: Checking installation
✓ The `go` executable was found on your system at: /usr/local/go/bin/go

-> Go: Checking minimum version requirements
✓ Your version of Go, 1.13.4, meets the minimum requirements.

-> Go: Checking Package Management
✓ You are using Go Modules (`go`) for package management.

-> Go: Checking PATH
✓ Your PATH contains /home/boon/gowork/bin.

-> Node: Checking installation
✓ The `node` executable was found on your system at: /usr/local/bin/node

-> Node: Checking minimum version requirements
✓ Your version of Node, v8.16.2, meets the minimum requirements.

-> NPM: Checking installation
✓ The `npm` executable was found on your system at: /home/boon/.npm-global/bin/npm

-> NPM: Checking minimum version requirements
✓ Your version of NPM, 6.2.0, meets the minimum requirements.

-> Yarn: Checking installation
✓ The `yarnpkg` executable was found on your system at: /usr/bin/yarnpkg

-> Yarn: Checking minimum version requirements
✘ Your version of Yarn, 1.3.2, does not meet the minimum requirements.

Minimum versions of Yarn are:

* >=1.12

For help setting up your Yarn environment please follow the instructions for you platform at:

https://yarnpkg.com/en/docs/install

-> PostgreSQL: Checking installation
✘ The `postgres` executable could not be found on your system.
For help setting up your Postgres environment please follow the instructions for you platform at:

https://www.postgresql.org/download/

-> MySQL: Checking installation
✓ The `mysql` executable was found on your system at: /usr/bin/mysql

-> MySQL: Checking minimum version requirements
✓ Your version of MySQL, 14.14, meets the minimum requirements.

-> SQLite3: Checking installation
✓ The `sqlite3` executable was found on your system at: /usr/bin/sqlite3

-> SQLite3: Checking minimum version requirements
✓ Your version of SQLite3, 3.22.0, meets the minimum requirements.

-> Cockroach: Checking installation
✓ The `cockroach` executable was found on your system at: /usr/local/bin/cockroach

-> Cockroach: Checking minimum version requirements
✓ Your version of Cockroach, 1.1.7, meets the minimum requirements.

-> Buffalo (CLI): Checking installation
✓ The `buffalo` executable was found on your system at: /home/boon/gowork/bin/buffalo

-> Buffalo (CLI): Checking minimum version requirements
✓ Your version of Buffalo (CLI), v0.15.5, meets the minimum requirements.

-> Buffalo: Application Details
Pwd         /home/boon/gowork/src/bitbucket.org/boyzfunhouse/weathermonitr
Root        /home/boon/gowork/src/bitbucket.org/boyzfunhouse/weathermonitr
GoPath      /home/boon/gowork
PackagePkg  bitbucket.org/boyzfunhouse/weathermonitr
ActionsPkg  bitbucket.org/boyzfunhouse/weathermonitr/actions
ModelsPkg   bitbucket.org/boyzfunhouse/weathermonitr/models
GriftsPkg   bitbucket.org/boyzfunhouse/weathermonitr/grifts
WithModules true
Name        weathermonitr
Bin         bin/weathermonitr
VCS         git
WithPop     true
WithSQLite  false
WithDep     false
WithWebpack true
WithNodeJs  true
WithYarn    true
WithDocker  true
WithGrifts  true
AsWeb       true
AsAPI       false
InApp       true
PackageJSON {map[build:webpack -p --progress dev:webpack --watch]}

-> Buffalo: config/buffalo-app.toml
name = "weathermonitr"
bin = "bin/weathermonitr"
vcs = "git"
with_pop = true
with_sqlite = false
with_dep = false
with_webpack = true
with_nodejs = true
with_yarn = true
with_docker = true
with_grifts = true
as_web = true
as_api = false

-> Buffalo: config/buffalo-plugins.toml
[[plugin]]
  binary = "buffalo-pop"
  go_get = "github.com/gobuffalo/buffalo-pop"

-> Buffalo: go.mod
module bitbucket.org/boyzfunhouse/weathermonitr

go 1.13

require (
	github.com/denisbrodbeck/machineid v1.0.1
	github.com/eclipse/paho.mqtt.golang v1.2.0
	github.com/fsnotify/fsnotify v1.4.8 // indirect
	github.com/gobuffalo/buffalo v0.15.5
	github.com/gobuffalo/buffalo-pop v1.23.1
	github.com/gobuffalo/envy v1.9.0
	github.com/gobuffalo/fizz v1.9.8 // indirect
	github.com/gobuffalo/flect v0.2.1 // indirect
	github.com/gobuffalo/gocraft-work-adapter v0.0.0-20180714213200-7d6504f1dffe
	github.com/gobuffalo/mw-csrf v1.0.0
	github.com/gobuffalo/mw-i18n v0.0.0-20190129204410-552713a3ebb4
	github.com/gobuffalo/mw-paramlogger v0.0.0-20190129202837-395da1998525
	github.com/gobuffalo/nulls v0.2.0
	github.com/gobuffalo/packr/v2 v2.8.0
	github.com/gobuffalo/plush v3.8.3+incompatible
	github.com/gobuffalo/pop v4.13.1+incompatible
	github.com/gobuffalo/suite v2.8.2+incompatible
	github.com/gobuffalo/tags v2.1.7+incompatible
	github.com/gobuffalo/validate v2.0.4+incompatible
	github.com/gocraft/work v0.5.1 // indirect
	github.com/gofrs/uuid v3.2.0+incompatible
	github.com/gomodule/redigo v2.0.0+incompatible
	github.com/google/uuid v1.1.1 // indirect
	github.com/gorilla/mux v1.7.4 // indirect
	github.com/hashicorp/hcl v1.0.0
	github.com/influxdata/influxdb v1.5.1-0.20200210134220-6b07ed72002c
	github.com/jackc/pgconn v1.4.0 // indirect
	github.com/markbates/grift v1.5.0
	github.com/markbates/refresh v1.11.1 // indirect
	github.com/mattn/go-colorable v0.1.6 // indirect
	github.com/pelletier/go-toml v1.6.0 // indirect
	github.com/pkg/errors v0.9.1
	github.com/robfig/cron v1.2.0 // indirect
	github.com/sirupsen/logrus v1.4.2
	github.com/unrolled/secure v0.0.0-20190103195806-76e6d4e9b90c
	github.com/utahta/go-linenotify v0.4.2
	golang.org/x/crypto v0.0.0-20200302210943-78000ba7a073
	golang.org/x/net v0.0.0-20200301022130-244492dfa37a // indirect
	golang.org/x/sys v0.0.0-20200302150141-5c8b2ff67527 // indirect
)

`
